lib/rubocop/cop/layout/class_structure.rb

```diff
@@ -289,12 +289,16 @@ def begin_pos_with_comment(node)
           (node.first_line - 1).downto(1) do |annotation_line|
             break unless (comment = processed_source.comment_at_line(annotation_line))
 
-            first_comment = comment
+            first_comment = comment if whole_line_comment_at_line?(annotation_line)
           end
 
           start_line_position(first_comment || node)
         end
 
+        def whole_line_comment_at_line?(line)
+          /^\s*#/.match?(processed_source.lines[line - 1])
```

Comment:
If my understanding is correct, it looks like there is no need for multi-line match.

```suggestion
          /\A\s*#/.match?(processed_source.lines[line - 1])
```

Reply:
No, that's just a bad habit of mine. I tend to use `^` even for single line matches. I'll change it.

Reply:
I did an `amend` with the suggested change to the commit and force-pushed. Didn't notice that there are convenient buttons here in GitHub. :slightly_smiling_face: